core/domain/learner_progress_services_test.py

```diff
@@ -481,6 +507,89 @@ def test_get_all_completed_exp_ids(self):
         self.assertEqual(learner_progress_services.get_all_completed_exp_ids(
             self.user_id), [self.EXP_ID_0, self.EXP_ID_1])
 
+    def test_get_filtered_completed_exp_summaries(self):
+        # Add explorations to the completed list.
+        learner_progress_services.mark_exploration_as_completed(
+            self.user_id, self.EXP_ID_0)
+        learner_progress_services.mark_exploration_as_completed(
+            self.user_id, self.EXP_ID_1)
+        learner_progress_services.mark_exploration_as_completed(
+            self.user_id, self.EXP_ID_3)
+        self.assertEqual(
+            len(learner_progress_services.get_all_completed_exp_ids(
+                self.user_id)), 3)
+
+        # Unpublish EXP_ID_3 to change status to ACTIVITY_STATUS_PRIVATE.
+        system_user = user_services.UserActionsInfo(feconf.SYSTEM_COMMITTER_ID)
+        rights_manager.unpublish_exploration(system_user, self.EXP_ID_3)
+        private_exploration = exp_services.get_exploration_summary_by_id(
+            self.EXP_ID_3)
+        self.assertEqual(
+            private_exploration.status, constants.ACTIVITY_STATUS_PRIVATE)
+
+        # Call get_activity_progress to get filtered progress.
+        user_activity = learner_progress_services.get_activity_progress(
+            self.user_id)
+        all_filtered_summaries = user_activity[0]
+        completed_exp_summaries = (
+            all_filtered_summaries.completed_exp_summaries)
+
+        # Test that completed exp summaries don't include private exploration.
+        # Ensure that completed_exp_summaries[0] matches EXP_ID_0.
+        self.assertEqual(
+            completed_exp_summaries[0].id, '0_en_arch_bridges_in_england')
+        # Ensure that completed_exp_summaries[1] matches EXP_ID_1.
+        self.assertEqual(
+            completed_exp_summaries[1].id, '1_fi_arch_sillat_suomi')
+        self.assertEqual(len(completed_exp_summaries), 2)
+
+    def test_publishing_private_exploration_is_filtered_as_complete(self):
+        # Add explorations to the completed list.
+        learner_progress_services.mark_exploration_as_completed(
+            self.user_id, self.EXP_ID_0)
+        learner_progress_services.mark_exploration_as_completed(
+            self.user_id, self.EXP_ID_1)
+        learner_progress_services.mark_exploration_as_completed(
+            self.user_id, self.EXP_ID_3)
+
+        # Unpublish EXP_ID_3 to change status to ACTIVITY_STATUS_PRIVATE.
+        system_user = user_services.UserActionsInfo(feconf.SYSTEM_COMMITTER_ID)
+        rights_manager.unpublish_exploration(system_user, self.EXP_ID_3)
+        private_exploration = exp_services.get_exploration_summary_by_id(
+            self.EXP_ID_3)
+        self.assertEqual(
+            private_exploration.status, constants.ACTIVITY_STATUS_PRIVATE)
+
```

Comment:
Ditto, assert there are only two completed exploration exist after this step.

Reply:
Sure.

Reply:
Actually, I cannot do this because there still are 3 completed explorations after making something private. 
Unpublishing an exploration / making it private does not change whether an exploration is completed or not. So, all 3 are still completed.

Reply:
The real differentiation comes into play when I call get_activity_progress where we will see EXP_ID_3 is no longer completed.

Reply:
So we expected `get_activity_progress` to return 2 explorations, can you assert this value around these lines?

Reply:
@DubeySandeep Yep, I already do this at the end of each test. I assert that either 2 or 3 explorations are filtered.

Reply:
@ezl-13 I don't see any assertion around these lines (After making exploration private and before publishing), have I missed something?

Reply:
@DubeySandeep Ah, I see what you mean. So each test should have 2 calls to get_activity_progress. I'll fix that!

Reply:
I feel like it would be redundant with the test above it, since I will be copy-ing and paste-ing all of it, but I will do it.

Reply:
Wait, just to check, which parts are you copy-and-pasting? We should definitely avoid wholesale copy/paste so I figured it might be worth double checking.

You should probably only be adding assert statements and nothing else.

Reply:
I believe that in order to check if get_activity_progress returns only 2 explorations after marking an exploration as private, I essentially am replicating the behavior in test_get_filtered_completed_exp_summaries.

Checking that there are only 2 explorations after marking an exploration as private is exactly all the code from "# Call get_activity_progress to get filtered progress" to "self.assertEqual(len(completed_exp_summaries), 2)"

Reply:
> check if get_activity_progress returns only 2 explorations after marking an exploration as private, I essentially am replicating the behavior in test_get_filtered_completed_exp_summaries.
In test_get_filtered_completed_exp_summaries that test is `endline verification` whereas in this test the same thing is `baseline verification`. I thinking assertion related code will be replicated and we should avoid replicate `actions` related code.